src/main/java/uk/gov/justice/digital/delius/service/CourtService.java

```diff
@@ -105,6 +106,19 @@ public Court getCourt(String code) {
         });
     }
 
+    public List<Court> getCourts() {
+        return courtRepository
+            .findAll()
+            .stream()
+            // filter out duplicates (see note above)
+            .filter(court -> court.getCourtId().equals(getMostLikelyCourt(court.getCode()).orElseThrow().getCourtId()))
```

Comment:
Do you need a `distinct()` after this?  It appears to replace courts with the most likely court, in which case they would be duplicated?

Reply:
Maybe not  - there's a test for it :)